packages/backend/src/queryBuilder.ts

```diff
@@ -245,11 +245,23 @@ export const buildQuery = ({
         return undefined;
     };
 
+    const tablesSqlWhere = [...joinedTables].reduce<string[]>(
```

Comment:
let's only apply the `sqlWhere` from the main table

Reply:
Are you sure about this ? don't we want to also filter joined tables if yoou add a sqlWhere filter to it ? specially once we introduce user attribute row level access...

Reply:
I _think_ so. It's easier to start including `sql_filter` from joins later if we change our mind than doing the opposite. So I think we should just have.

In any case, I think it would get confusing because joins are only conditionally included in the query (depending on the filters selected) and those joined tables could have their own `sql_filter` referencing other join

Reply:
I imagine you can have  a schema like this 

```
models: 
  - name: users
    meta: 
      sql_filter: "${ld.user.user_id} = ${ld.user_id}"
  - companies: 
    meta: 
        join_on: 'users.company_id = companies.company_id'
```

This is configured so users can only see their own users (or any other attribute that will give them access) . However if you select from `company` while joining users, that filter will no longer apply, so any user can see all other users in the joined result.

Reply:
Like this ? 

![image](https://github.com/lightdash/lightdash/assets/1983672/9b794078-eb11-4d23-ab7e-2708c8a84038)

My `customers` filter is ignored
```
  - name: customers
    meta:
      sql_where: '(${first_name} IS NOT NULL AND ${last_name} IS NOT NULL)'
``

Reply:
I think right now it's confusing because each model serves two purposes:
1. as a table that can be joined to other tables (which ignores things like `join` or `hidden` or `group_label`) but includes it's dimensions metrics
2. as a base table, which joins other tables (including all the extra metadata)

So I'm thinking of `sql_filter` as something that is only applicable to when you use it as a base table.

Reply:
Actually @rephus 

If I join model A to model B. and model C to model B. 

Can model A reach model C?

Reply:
you're right, this #2049 can be a problem when joining multiple tables. 

In fact, even the compile doesn't like it. 

eg: I added a new model ( a copy of events)  only joined with payments 

![image](https://github.com/lightdash/lightdash/assets/1983672/0f543d02-0908-470d-b2d0-b89a2117725d)

I got this compile error

> - ERROR> joined : Model "payments" has a dimension reference: ${orders.amount} which matches no dimension

Reply:
Yep let's only do the base table.

Instead of #2049 - we might also make the yaml clearer so it's obvious when you're configuring the underlying table or an "explore" which is a bunch of joined tables with extra config (like `sql_where` and access)